fakestorage/object.go

```diff
@@ -24,24 +24,174 @@ var errInvalidGeneration = errors.New("invalid generation ID")
+func (o Object) MarshalJSON() ([]byte, error) {
+	temp := struct {
+		BucketName      string            `json:"bucket"`
+		Name            string            `json:"name"`
+		ContentType     string            `json:"contentType"`
+		ContentEncoding string            `json:"contentEncoding"`
+		Content         []byte            `json:"-"`
+		Crc32c          string            `json:"crc32c,omitempty"`
+		Md5Hash         string            `json:"md5Hash,omitempty"`
+		ACL             []ACLRule         `json:"acl,omitempty"`
+		Created         time.Time         `json:"created,omitempty"`
+		Updated         time.Time         `json:"updated,omitempty"`
+		Deleted         time.Time         `json:"deleted,omitempty"`
+		Generation      int64             `json:"generation,omitempty,string"`
+		Metadata        map[string]string `json:"metadata,omitempty"`
+	}{
+		BucketName:      o.BucketName,
+		Name:            o.Name,
+		ContentType:     o.ContentType,
+		ContentEncoding: o.ContentEncoding,
+		Content:         o.Content,
+		Crc32c:          o.Crc32c,
+		Md5Hash:         o.Md5Hash,
+		Created:         o.Created,
+		Updated:         o.Updated,
+		Deleted:         o.Deleted,
+		Generation:      o.Generation,
+		Metadata:        o.Metadata,
+	}
+	temp.ACL = make([]ACLRule, len(o.ACL))
+	for i, ACL := range o.ACL {
+		temp.ACL[i] = ACLRule(ACL)
+	}
+	return json.Marshal(temp)
+}
+
+// UnmarshalJSON for Object to use ACLRule instead of storage.ACLRule
+func (o *Object) UnmarshalJSON(data []byte) error {
+	temp := struct {
+		BucketName      string            `json:"bucket"`
+		Name            string            `json:"name"`
+		ContentType     string            `json:"contentType"`
+		ContentEncoding string            `json:"contentEncoding"`
+		Content         []byte            `json:"-"`
+		Crc32c          string            `json:"crc32c,omitempty"`
+		Md5Hash         string            `json:"md5Hash,omitempty"`
+		ACL             []ACLRule         `json:"acl,omitempty"`
+		Created         time.Time         `json:"created,omitempty"`
+		Updated         time.Time         `json:"updated,omitempty"`
+		Deleted         time.Time         `json:"deleted,omitempty"`
+		Generation      int64             `json:"generation,omitempty,string"`
+		Metadata        map[string]string `json:"metadata,omitempty"`
+	}{}
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	o.BucketName = temp.BucketName
+	o.Name = temp.Name
+	o.ContentType = temp.ContentType
+	o.ContentEncoding = temp.ContentEncoding
+	o.Content = temp.Content
+	o.Crc32c = temp.Crc32c
+	o.Md5Hash = temp.Md5Hash
+	o.Created = temp.Created
+	o.Updated = temp.Updated
+	o.Deleted = temp.Deleted
+	o.Generation = temp.Generation
+	o.Metadata = temp.Metadata
+	o.ACL = make([]storage.ACLRule, len(temp.ACL))
+	for i, ACL := range temp.ACL {
+		o.ACL[i] = storage.ACLRule(ACL)
+	}
+
+	return nil
+}
+
+// ACLRule is an alias of storage.ACLRule to have custom JSON marshal
+type ACLRule storage.ACLRule
```

Comment:
Let's not export this type since it's only used for marshaling/unmarshaling.
```suggestion
type aclRule storage.ACLRule
```